nominal/core/client.py

```diff
@@ -511,6 +512,51 @@ def get_connection(self, rid: str) -> Connection:
         response = self._clients.connection.get_connection(self._clients.auth_header, rid)
         return Connection._from_conjure(self._clients, response)
 
+    def create_video_from_mcap_io(
+        self,
+        mcap: BinaryIO,
+        topic: str,
+        name: str,
+        description: str | None = None,
+        file_type: tuple[str, str] | FileType = FileTypes.BINARY,
+        *,
+        labels: Sequence[str] = (),
+        properties: Mapping[str, str] | None = None,
+    ) -> Video:
+        """Create videos from topics in a mcap file.
+
+        Mcap must be a file-like object in binary mode, e.g. open(path, "rb") or io.BytesIO.
+
+        If name is None, the name of the file will be used.
+        """
+        if isinstance(mcap, TextIOBase):
+            raise TypeError(f"dataset {mcap} must be open in binary mode, rather than text mode")
+
+        file_type = FileType(*file_type)
+        urlsafe_name = urllib.parse.quote_plus(name)
+        filename = f"{urlsafe_name}{file_type.extension}"
+
+        s3_path = put_multipart_upload(
+            self._clients.auth_header, mcap, filename, file_type.mimetype, self._clients.upload
+        )
+        request = ingest_api.IngestMcapRequest(
+            channel_config=[
+                ingest_api.McapChannelConfig(
+                    channel_type=ingest_api.McapChannelConfigType(video=ingest_api.McapVideoChannelConfig()),
+                    locator=api.McapChannelLocator(topic=topic),
+                )
+            ],
+            labels=list(labels),
+            properties={} if properties is None else dict(properties),
+            sources=[ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path))],
+            description=description,
+            title=name,
+        )
+        response = self._clients.ingest.ingest_mcap(self._clients.auth_header, request)
+        video_rid = response.outputs[0].target.video_rid
+        assert video_rid is not None, "No RID returned"
```

Comment:
let's handle this a bit more explicitly:
- check the length of outputs and ensure it is exactly 1 (so users don't see an IndexError, and in case there's somehow >1)
- check and raise an exception rather than assert (so that users can handle the exception reliably) - this one is mostly annoying because handling the conjure unions is kinda silly, but we don't have a great pattern for handling this across the codebase yet. it won't ever be none if the `target.type == "video"` (the constructor itself raises if that happens)